lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -365,25 +365,54 @@ def set_inhibit_warnings_for_pod(pod_name, should_inhibit)
 
       #--------------------------------------#
 
-      # Sets whether the target definition should build a framework.
+      # The (desired) build type for the pods integrated in this target definition. Defaults to static libraries and can
+      # only be overridden through Pod::Podfile::DSL#use_frameworks!.
+      #
+      # @return [BuildType]
+      #
+      def build_type
+        value = get_hash_value('uses_frameworks', root? ? BuildType.static_library : parent.build_type)
+        case value
+        when true, false
+          value ? BuildType.dynamic_framework : BuildType.static_library
+        when Hash
+          BuildType.new(:linkage => value.fetch(:linkage), :packaging => value.fetch(:packaging))
```

Comment:
will this work with string-based keys when used instead of symbols? not sure if we normalize this or not

Reply:
oh you mean `use_frameworks! 'linkage' => :dynamic`?

Reply:
I am a bit confused can you show an example of when this can happen?

Reply:
Yeah just like you mentioned above, or if parsing a YAML file where the key was “linkage” and not “:linkage”

Reply:
There is no prior values in YAML files that have serialized `use_frameworks!` as a hash since the feature does not exist. The only way it is serialized today is as a boolean true or false.

I also don't think we want to allow `use_frameworks! 'linkage' => :dynamic` currently.

Reply:
yes but YAML isn't always generated, we support hand-written YAML. 

why don't we want to allow something like `use_frameworks! 'linkage' => 'dynamic'`? not all CocoaPods users are super familiar with Ruby and symbols can be confusing

Reply:
I dont have a great answer as to why we don't allow it besides I don't think there is any prior precedent established here that shows CocoaPods allowing strings for keys or values. The only place this occurs is when we serialize podspecs to JSON which do not support symbols. That happens behind the scenes and its not "user visible".

Reply:
makes sense. FWIW looks like we allow it on installation options, but don't allow it in some other places

```ruby
install 'cocoapods', 'disable_input_output_paths' => true
```

Reply:
interesting! you are right and it stores Strings as its keys:
```ruby
      def self.option(name, default, boolean: true)
        name = name.to_s
        raise ArgumentError, "The `#{name}` option is already defined" if defaults.key?(name)
        defaults[name] = default
        attr_accessor name
        alias_method "#{name}?", name if boolean
      end
```

Reply:
I'll proceed without it for now...........